realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -306,17 +366,17 @@ boolean addAllInternal(Collection<? extends Integer> c) {
     }
 
     @Override
-    boolean retainAllInternal(Collection<?> c) {
+    boolean removeAllInternal(Collection<?> c) {
```

Comment:
Remember to check the type. Might have been guaranteed somewhere up the hierarchy, but lost track of it. If that is in fact the case, then put a command about why the unchecked cast is safe.  

Similarly on the other methods.